File: Blind 75 LeetCode/Pankajavalli RV/Maximum Subarray/code.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
      
        maxSum = nums[0]
        currentSum = nums[0]
      
        for n in nums[1:]:
            currentSum = max(n, currentSum + n)
            maxSum = max(maxSum, currentSum)
          
        return maxSum

File: Blind 75 LeetCode/Sakshi Rai/Maximum Subarray/code.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_sum=INT_MIN;
        int curr_sum=0;

        for(int i=0;i<nums.size();i++){
            curr_sum=max(nums[i],curr_sum+nums[i]);
            max_sum=max(max_sum,curr_sum);
        }
        return max_sum;
    }
};
